src/com/t_oster/liblasercut/drivers/K40NanoRasterProperty.java

```diff
@@ -0,0 +1,166 @@
+/**
+ * This file is part of LibLaserCut.
+ * Copyright (C) 2011 - 2014 Thomas Oster <mail@thomas-oster.de>
+ *
+ * LibLaserCut is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * LibLaserCut is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with LibLaserCut. If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+package com.t_oster.liblasercut.drivers;
+
+import com.t_oster.liblasercut.LaserProperty;
+
+/**
+ *
+ * @author Tat
+ */
+public class K40NanoRasterProperty implements LaserProperty
+{
+  
+  static final String VAR_MM_PER_SECOND = "mm per second";
+  static final String VAR_RASTER_STEP = "raster step";

```

Comment:
I'm not sure about the purpose of "raster step", which seems to be a division factor to reduce the engrave DPI.
All other drivers just use the engrave DPI, which can be changed in VisiCut (slightly awkwardly via options - laser profiles - edit, but that's a different issue).

Reply:
It is, but it's oddly a division that occurs within the board. When in raster mode the speedcode has a G parameter it physically steps that number of mils when the right-left or top-bottom flag switches.  I added it because it was there, not sure how visicut does things which seems to be why the raster as a whole doesn't work. The K40 generally only has 1000 DPI, because it has stepper motors that step 1 mil. That property there makes the stepper motor step down by more than 1 row.  It might not be applicable.